pkg/virt-controller/watch/snapshot.go

```diff
@@ -637,6 +637,46 @@ func (ctrl *SnapshotController) updateSnapshotStatus(vmSnapshot *snapshotv1.Virt
 	return nil
 }
 
+func (ctrl *SnapshotController) updateVolumeSnapshotStatuses(vm *kubevirtv1.VirtualMachine) error {
+	log.Log.V(3).Infof("Update volume snapshot status for VM [%s/%s]", vm.Namespace, vm.Name)
+
+	var statuses []kubevirtv1.VolumeSnapshotStatus
+	for _, volume := range vm.Spec.Template.Spec.Volumes {
+		log.Log.V(3).Infof("Update volume snapshot status for volume [%s]", volume.Name)
+		status := ctrl.getVolumeSnapshotStatus(vm, &volume)
+		statuses = append(statuses, status)
+	}
+
+	vm.Status.VolumeSnapshotStatuses = statuses
+
+	return nil
+}
+
+func (ctrl *SnapshotController) getVolumeSnapshotStatus(vm *kubevirtv1.VirtualMachine, volume *kubevirtv1.Volume) kubevirtv1.VolumeSnapshotStatus {
+	pvc, err := ctrl.getSnapshotPVC(vm.Namespace, volume.Name)
```

Comment:
Volume name is not PVC name

Reply:
This is confusing: `getSnapshotPVC(namespace string, volumeName string)`.

Anyway, I realized this is probably wrong. This gets PVCs for an existing snapshot, which is different from the information we need, which is: is snapshotting even possible.